src/main/java/com/code_intelligence/jazzer/mutation/mutator/collection/SetMutatorFactory.java

```diff
@@ -0,0 +1,207 @@
+import static com.code_intelligence.jazzer.mutation.support.TypeSupport.parameterTypesIfParameterized;
+import static java.lang.Math.min;
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toSet;
+
+import com.code_intelligence.jazzer.mutation.annotation.WithSize;
+import com.code_intelligence.jazzer.mutation.api.Debuggable;
+import com.code_intelligence.jazzer.mutation.api.ExtendedMutatorFactory;
+import com.code_intelligence.jazzer.mutation.api.MutatorFactory;
+import com.code_intelligence.jazzer.mutation.api.PseudoRandom;
+import com.code_intelligence.jazzer.mutation.api.SerializingInPlaceMutator;
+import com.code_intelligence.jazzer.mutation.api.SerializingMutator;
+import com.code_intelligence.jazzer.mutation.support.RandomSupport;
+import com.code_intelligence.jazzer.mutation.support.StreamSupport;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.AnnotatedType;
+import java.util.LinkedHashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+final class SetMutatorFactory implements MutatorFactory {
+  @Override
+  public Optional<SerializingMutator<?>> tryCreate(
+      AnnotatedType type, ExtendedMutatorFactory factory) {
+    return parameterTypesIfParameterized(type, Set.class)
+        .map(
+            parameterTypes ->
+                parameterTypes.stream()
+                    .map(innerType -> propagatePropertyConstraints(type, innerType))
+                    .map(factory::tryCreate)
+                    .flatMap(StreamSupport::getOrEmpty)
+                    .collect(Collectors.toList()))
+        .filter(elementMutator -> elementMutator.size() == 1)
+        .map(
+            elementMutator -> {
+              int min = SetMutator.DEFAULT_MIN_SIZE;
+              int max = SetMutator.DEFAULT_MAX_SIZE;
+              for (Annotation annotation : type.getDeclaredAnnotations()) {
+                if (annotation instanceof WithSize) {
+                  WithSize withSize = (WithSize) annotation;
+                  min = withSize.min();
+                  max = withSize.max();
+                }
+              }
+
+              return new SetMutator<>(elementMutator.get(0), min, max);
+            });
+  }
+
+  private static final class SetMutator<K> extends SerializingInPlaceMutator<Set<K>> {
+    private static final int DEFAULT_MIN_SIZE = 0;
+    private static final int DEFAULT_MAX_SIZE = 1000;
+
+    private final SerializingMutator<K> keyMutator;
+    private final int minSize;
+    private final int maxSize;
+
+    SetMutator(SerializingMutator<K> keyMutator, int minSize, int maxSize) {
+      this.keyMutator = keyMutator;
+      this.minSize = Math.max(minSize, DEFAULT_MIN_SIZE);
+      this.maxSize = Math.min(maxSize, DEFAULT_MAX_SIZE);
```

Comment:
Do we need the clamping with `DEFAULT_MIN/MAX_SIZE`? The `require`s below already perform validation on the parameters and without the clamping it would be possible to set `maxSize > DEFAULT_MAX_SIZE` which could be useful in some cases.